lib/symboldatabase.cpp

```diff
@@ -2474,6 +2476,20 @@ const Token * Function::constructorMemberInitialization() const
     return nullptr;
 }
 
+bool Function::isBodyEmpty() const
+{
+    if (!hasBody())
+    {
+        return true;
+    }
+    if (!functionScope || !functionScope->bodyStart || !functionScope->bodyStart->next())
```

Comment:
hmm.. if `hasBody()` is true the I expect that there is a bodyStart.

Reply:
and every scope has a bodyStart and bodyEnd as far as I know.

Reply:
`functionScope` and 'functionScope->bodyStart` are pointers, so it's just a double-check to avoid segfault. I can remove it if you insist.